### ISSUE TITLE
bug: Error when getting output type of "require('discord.js')"

### ISSUE THREAD

**Issue by BannerBomb** [sapphiredev/type#547]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

Only seems to happen when requiring discord.js and evaluating the output of it using the Type class. So I doubt you would want to fix it. Just thought I'd make a bug report just in case.

Example code includes
```js
const { Type } = require('@sapphire/type');
const DJS = require('discord.js');
const output = new Type(DJS);
output.toString();
```
Will cause the error.

```ruby
TypeError: Cannot read properties of null (reading 'name')
     at Function.resolve (/opt/stratobotelite/node_modules/@sapphire/type/src/lib/index.ts:152:2)
     at new Type (/opt/stratobotelite/node_modules/@sapphire/type/src/lib/index.ts:60:13)
     at _Type.addValue (/opt/stratobotelite/node_modules/@sapphire/type/src/lib/index.ts:102:25)
     at _Type.addEntry (/opt/stratobotelite/node_modules/@sapphire/type/src/lib/index.ts:112:30)
     at _Type.check (/opt/stratobotelite/node_modules/@sapphire/type/src/lib/index.ts:134:2)
     at _Type.toString (/opt/stratobotelite/node_modules/@sapphire/type/src/lib/index.ts:77:7)
     at Function.codeBlock (/opt/stratobotelite/src/frameworks/klasa/lib/util/util.js:41:42)
     at module.exports.run (/opt/stratobotelite/src/bot/commands/Admin/eval.js:138:23)
     at processTicksAndRejections (node:internal/process/task_queues:95:5)
     at async module.exports.runCommand (/opt/stratobotelite/src/bot/monitors/commandHandler.js:75:23)
```

### Steps To Reproduce

```js
const { Type } = require('@sapphire/type');
const DJS = require('discord.js');
const output = new Type(DJS);
output.toString();
```

### Expected behavior

Wasn't expected to error.

### Screenshots

![image](https://github.com/sapphiredev/type/assets/11788894/8142202a-c703-4a11-9366-899c5ca0f36b)
![image](https://github.com/sapphiredev/type/assets/11788894/55c18d34-38ad-434f-9e5f-d5c87d4d7ab9)

### System Info

```shell
System:
    OS: Linux 4.15 Debian GNU/Linux 12 (bookworm) 12 (bookworm)
    CPU: (4) x64 Intel Core Processor (Haswell, no TSX)
    Memory: 6.34 GB / 7.60 GB
    Container: Yes
    Shell: 5.2.15 - /bin/bash
  Binaries:
    Node: 21.7.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.5.0 - /usr/local/bin/npm
```

**Comment by favna** [comment]
This is not the intended use case of this package. It's supposed to be used on a variable, not on a module.

**Comment by BannerBomb** [comment]
> This is not the intended use case of this package. It's supposed to be used on a variable, not on a module. That said it is odd. We can look into it.

yeah, I just made a pull request because if it happens on something like this it may error in certain circumstances.

So I did a test and these are the modules that it errors on when trying to read them.
```js
[
  [ 'SlashCommandBuilder', [Function: SlashCommandBuilder] ],
  [
    'SlashCommandChannelOption',
    [Function: SlashCommandChannelOption]
  ],
  [
    'SlashCommandIntegerOption',
    [Function: SlashCommandIntegerOption]
  ],
  [ 'SlashCommandNumberOption', [Function: SlashCommandNumberOption] ],
  [ 'SlashCommandStringOption', [Function: SlashCommandStringOption] ],
  [
    'SlashCommandSubcommandBuilder',
    [Function: SlashCommandSubcommandBuilder]
  ],
  [
    'SlashCommandSubcommandGroupBuilder',
    [Function: SlashCommandSubcommandGroupBuilder]
  ]
]
```

Found by doing
```js
s!eval const { Type } = require('@sapphire/type');
const results = [];
const errors = [];
for (const [key, value] of Object.entries(require('discord.js'))) {
    try {
        results.push(new Type(value));
    } catch (error) {
        results.push(error);
        errors.push([key, value])
    }
}
console.log(errors);
```

**Comment by favna** [comment]
Released in v2.5.1
